currency/l1/public.yml

```diff
@@ -159,3 +159,102 @@ paths:
             text/html:
               schema:
                 type: string
+
+  /validate:
+    post:
+      tags:
+        - Currency
+      description: Validate signed data update.
+      requestBody:
+        description: Signed data update.
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '../../models/schemas/SignedDataUpdate.yml'
+      responses:
+        204:
+          description: Data update is valid.
+          headers:
+            X-Id:
+              $ref: '../../models/headers/X-Id.yml'
+            X-Session-Token:
+              $ref: '../../models/headers/X-Session-Token.yml'
+            Request-Signature:
+              $ref: '../../models/headers/Request-Signature.yml'
+        503:
+          description: Last Global Snapshot ordinal not available, or invalid data update.
```

Comment:
`503` is perfect for `Last Global Snapshot ordinal not available` but I don't think it's a good error code for invalid data update. `503` specifies that the server cannot handle the request right now (which is the case when the global snapshot is not yet initialized).

Reply:
@marcinwadon FYI that's how it was before. I agree a 4xx error is more appropriate. Here are some suggestions:

1. [422 Unprocessable Content](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/422) ([WebDAV](https://developer.mozilla.org/en-US/docs/Glossary/WebDAV))
The request was well-formed but was unable to be followed due to semantic errors.
2. [406 Not Acceptable](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/406)
This response is sent when the web server, after performing [server-driven content negotiation](https://developer.mozilla.org/en-US/docs/Web/HTTP/Content_negotiation#server-driven_content_negotiation), doesn't find any content that conforms to the criteria given by the user agent.
3. [400 Bad Request](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/400)
The server cannot or will not process the request due to something that is perceived to be a client error (e.g., malformed request syntax, invalid request message framing, or deceptive request routing).

Reply:
These look like two different errors to me. If the update is invalid, then 400 is appropriate. If there's some internal issue with the node that makes the request unprocessible, then 500. 

503 would usually indicate that the service itself isn't available for some reason.

Reply:
Either 400: invalid update
or 503: last snapshot not available